File: RUBY/DataStructures/map.rb
class Map
    def initialize
        @my_array = [];
    end

    def set(key, val)
        if self.get(key)
            my_array.each do |arr|
                arr[1] = val if arr[0] == key
            end
        else
            my_array.push([key, val])
        end
    end

    def get(key)
        my_array.each do |arr|
            return arr[1] if arr[0] == key
        end
        return false;
    end

    def delete(key)
        if self.get(key)
            my_array.each_with_index do |arr, i|
                my_array[i] = "" if arr[0] == key
            end
            rocky = my_array.select{|arr| arr != ""}
            self.my_array = rocky
        else
            return "Not found"
        end
    end

    def show
        p my_array
    end

    private

    attr_accessor :my_array
end

bop = Map.new
bop.set(1,2)
bop.set(2,5)
bop.set(9,8)
bop.show
bop.delete(2)
bop.show

File: JavaScript/DataStructures/map.js
function Map() {
    this.array = [];
}

Map.prototype.set = function(key, value) {
    if(this.get(key)) {
        for(let i = 0; i < this.array.length; i++) {
            if(this.array[i][0] === key) {
                this.array[i][1] = value;
            }
        }
    } else {
        this.array.push([key, value]);
    }
}

Map.prototype.get = function(key) {
    for(let i = 0; i < this.array.length; i++) {
        if(this.array[i][0] === key) {
            return this.array[i][1]
        }
    }
    return false
}

Map.prototype.delete = function(key) {
    if(this.get(key)) {
        for(let i = 0; i < this.array.length; i++) {
            if(this.array[i][0] === key) {
                this.array[i] = ""
            }
        }
        let arr = [];
        for(let j = 0; j < this.array.length; j++) {
            if(this.array[j] !== "") {
                arr.push(this.array[j])
            }
        }
        this.array = arr;
    } else {
        return "Not found";
    }
}

Map.prototype.show = function() {
    console.log(this.array);
}

let feds = new Map();
feds.set(1,2);
feds.set(3,4);
feds.set(5,6);
feds.show();
feds.delete(3);
feds.show();